apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -194,4 +196,118 @@ defmodule Snitch.Domain.Order.Transitions do
         {:halt, error}
     end)
   end
+
+  # Guard Class to check negative amount is skipped.
+
+  defp calc_payable_amount(order) do
+    packages_total = Package.compute_packages_total(order)
+    total_amount = Money.add!(packages_total, order.total)
+    amount_paid_before = Payment.compute_order_payments(order)
+    Money.sub!(total_amount, amount_paid_before)
+  end
+
+  defp process_payment_chk(context, multi, payment_method, payment, order, amount_to_pay) do
+    params = %{
+      amount: amount_to_pay,
+      order_id: order.id,
+      slug: "check-payment"
+    }
+
+    function = fn _ ->
+      CheckPaymentModel.create(params)
+    end
+
+    struct(context, multi: Multi.run(multi, :checkpayment, function))
+  end
+
+  defp process_payment_ccd(context, multi, payment_method, payment, order, amount_to_pay) do
+    case validate_payment_method(payment) do
+      %{changes: %{card: %{errors: []}}} ->
+        params = %{
+          amount: amount_to_pay,
+          payment_type: "ccd",
+          payment_method_id: payment_method.id
+        }
+
+        function = fn _ ->
+          CardPaymentModel.create("card-payment", order.id, params, %{card: payment})
+        end
+
+        struct(context, multi: Multi.run(multi, :cardpayment, function))
+
+      %{changes: %{card: %{errors: errors}}} ->
+        struct(context, valid?: false, errors: errors)
+    end
+  end
+
+  @doc """
+  Validator for payment method
+
+  Responses
+  {:ok, payment}
+  {:error, errors}
+  """
+
+  defp validate_payment_method(payment) do
+    CardPayment.create_changeset(%CardPayment{}, %{
+      card: payment
+    })
+  end
+
+  @doc """
+  Processing incoming payment.
+
+    ## Schema of the `:state`
+
+    ### card
+
+    ```
+    %{
+      payment:
+        %{
+          payment_method_id: "payment_method_id",
+          month: 12,
+          name_on_card: "Helper",
+          brand: "VISA",
+          number: "411141114114111",
+          card_name: "Rupay",
+          user_id: "Somer user id"
+        }
+    }
+
+    `user_id` needs to be injected into map through webcontoller.
+
+    ### Check
+    %{
+      payment:
+        %{
+          payment_method_id: "payment"
+        }
+    }
+    ```
+
+    ### 3rd party
+
+    * Payubiz
+
+    
+
+  """
+  @spec compute_order_payment(Context.t()) :: Context.t()
+  def compute_order_payment(
+        %Context{valid?: true, struct: %Order{id: order_id} = order} = context
+      ) do
+    %{state: %{payment: payment}, multi: multi} = context
+
+    amount_to_pay = calc_payable_amount(order)
+    payment_method = PaymentMethod.get(payment.payment_method_id)
+
+    case payment_method.code do
```

Comment:
It is much cleaner to make a function with 2 clauses to handle this, since:
1. you've already made 2 functions
2. the `case` can be removed